src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -2044,7 +2072,22 @@ private async ValueTask<JsonRpcMessage> DispatchIncomingRequestAsync(JsonRpcRequ
                     }
                 }
 
-                return await this.DispatchRequestAsync(request, targetMethod, cancellationToken).ConfigureAwait(false);
+                if (this.JoinableTaskFactory is null || request.TryGetTopLevelProperty<string>(JoinableTaskTokenHeaderName, out string? parentToken) is false || parentToken is null)
```

Comment:
We can use our own `AsyncLocal<string?>` to propagate the token even without a JTF instance.